be/src/exec/avro_scanner.cpp

```diff
@@ -0,0 +1,525 @@
+                                       avro::Decoder& decoder, int nullcount) mutable {
+            double val_double = decoder.decodeDouble();
+            wbytes = sprintf((char*)tmp_buf, "%lf", val_double);
+            _fill_slot(tuple, slot_desc, tuple_pool, tmp_buf, wbytes);
+        };
+    } break;
+    case avro::AVRO_BOOL: {
+        return [tmp_buf, wbytes, this](MemPool* tuple_pool, Tuple* tuple, SlotDescriptor* slot_desc,
+                                       avro::Decoder& decoder, int nullcount) mutable {
+            bool val_bool = decoder.decodeBool();
+            wbytes = sprintf((char*)tmp_buf, "%d", val_bool);
+            _fill_slot(tuple, slot_desc, tuple_pool, tmp_buf, wbytes);
+        };
+    } break;
+    case avro::AVRO_ARRAY: {
+        // should have array type info
+        // TODO :
+    } break;
+    case avro::AVRO_UNION: {
+        auto nullable_deserializer = [root_node, slot_desc, this](size_t non_null_union_index) {
+            auto nested_deserialize =
+                    createDeserializeFn(root_node->leafAt(non_null_union_index), slot_desc);
+            return [non_null_union_index, nested_deserialize, this](
+                           MemPool* tuple_pool, Tuple* tuple, SlotDescriptor* slot_desc,
+                           avro::Decoder& decoder, int nullcount) mutable {
+                size_t union_index = decoder.decodeUnionIndex();
+                if (union_index == non_null_union_index) {
+                    nested_deserialize(tuple_pool, tuple, slot_desc, decoder, nullcount);
+                } else {
+                    tuple->set_null(slot_desc->null_indicator_offset());
+                }
+            };
+        };
+        if (root_node->leaves() == 2 && slot_desc->is_nullable()) {
+            if (root_node->leafAt(0)->type() == avro::AVRO_NULL) return nullable_deserializer(1);
+            if (root_node->leafAt(1)->type() == avro::AVRO_NULL) return nullable_deserializer(0);
+        }
+    } break;
+    case avro::AVRO_NULL: {
+        if (slot_desc->is_nullable()) {
+            return [this](MemPool* tuple_pool, Tuple* tuple, SlotDescriptor* slot_desc,
+                          avro::Decoder& decoder, int nullcount) mutable {
+                tuple->set_null(slot_desc->null_indicator_offset());
+                nullcount++;
+            };
+        } else {
+            throw avro::Exception("NULL data for non-nullable column" + slot_desc->col_name());
+        }
+    } break;
+    case avro::AVRO_ENUM:
+        [[fallthrough]];
+    case avro::AVRO_FIXED:
+        [[fallthrough]];
+    case avro::AVRO_MAP:
+        [[fallthrough]];
+    case avro::AVRO_RECORD:
+        [[fallthrough]];
+    default: {
+        throw avro::Exception("Not support " + root_node->type() + std::string("type yet."));
+    } break;
+    }
+    throw avro::Exception("Type " + slot_desc->type().type +
+                          std::string("is not compatible with Avro ") +
```

Comment:
```suggestion
                          std::string(" is not compatible with Avro ") +
```

Reply:
Thx for your advice~ Fixed all the comments above.